### DISCUSSION TITLE
FindMarkers returns avg_log2FC with inf or -inf for some genes, is it normal?

### DISCUSSION THREAD

**Discussion by lkqnaruto** [satijalab/seurat#4499] [Q&A] [answered]
Hi

I'm running FindMarkers to compare 1 cluster but for different group, and in the return, some genes have avg_log2FC with inf, and some have -inf, is it normal? Sorry, I'm not a biologist.

**Comment by timoast** [comment, marked as answer]
If one group of cells being compared had 0 counts, then the fold change will be infinite

**Reply by lkqnaruto** [reply]
Thank you for the reply, if one group has 0 counts, then why both pct.1 and pct.2 are very close to one? From my understanding (please correct me if I am wrong), pct.1 and pct.2 represents the percentage of cells in each group being detected the current genes. For example, 99.5% of cells in group 1 detect `Lyz2`, and 100% of cells in group2 detect `Lyz2`. how come one of 2 groups has 0 counts? 

```
                       p_val  avg_log2FC pct.1 pct.2     p_val_adj
Lyz2            0.000000e+00        -Inf 0.995 1.000  0.000000e+00
COX3            2.496048e-03        -Inf 0.994 0.983  1.000000e+00
AC134224.3      1.515003e-06 -104.451534 0.990 0.993  2.895927e-02
COX2            4.203931e-38        -Inf 0.989 0.975  8.035813e-34
```

**Reply by timoast** [reply]
What assay are you running differential expression on? The fold change values seem very large
